packages/hooks/src/use-mutation-observer/use-mutation-observer.ts

```diff
@@ -0,0 +1,100 @@
+// NOTE: This React hook is essentially the same as the [@react-hook/intersection-observer](https://www.npmjs.com/package/@react-hook/intersection-observer)
+// except for the usage of `MutationObserver` instead of `IntersectionObserver`.
+import useLatest from '@react-hook/latest';
+import rafSchd from 'raf-schd';
+import { useLayoutEffect } from 'react';
+
+type TUseMutationObserverCallback = (
+  mutationsList: MutationRecord[],
+  observer: MutationObserver
+) => void;
+
+let _mutationObserver: ReturnType<typeof createMutationObserver>;
+
+function createMutationObserver() {
+  const callbacks = new Map<Node, TUseMutationObserverCallback[]>();
+  const observer = new MutationObserver(
+    rafSchd((mutationsList, observer) => {
+      const mutationsByTarget = mutationsList.reduce<
+        Map<Node, MutationRecord[]>
+      >((_mutationsByTarget, mutation) => {
+        const callbacksForTarget =
+          _mutationsByTarget.get(mutation.target) ?? [];
+        callbacksForTarget.push(mutation);
+        _mutationsByTarget.set(mutation.target, callbacksForTarget);
+        return _mutationsByTarget;
+      }, new Map());
+
+      mutationsByTarget.forEach((mutations, target) => {
+        const targetCallbacks = callbacks.get(target);
+        targetCallbacks?.forEach((cb) => cb(mutations, observer));
+      });
+    })
+  );
+
+  return {
+    observer,
+    subscribe(
+      target: HTMLElement,
+      callback: TUseMutationObserverCallback,
+      options?: MutationObserverInit
```

Comment:
We need to receive the options per `subscribe` call so we can have observe different elements with different configurations in the same observer.